doc/md/fundamentals/4-control-flow/2-loops.md

```diff
@@ -0,0 +1,103 @@
+---
+sidebar_position: 2
+---
+
+# Loops
+
+In Motoko, loops provide flexible control over repetition, such as iterating over collections, waiting for conditions to change, or running code indefinitely until explicitly stopped.
+
+Motoko supports different types of loops:
+
+- Unconditional loops: Run indefinitely unless manually stopped.
+
+- `for` loops: Iteration over collections.
+
+- `while` loops: Condition-based repetition.
+
+## Unconditional loops
+
+An unconditional loop runs indefinitely until it is explicitly stopped. Unlike `while` or `for` loops, which rely on a condition to determine when to exit, unconditional loops continue executing without any predefined exit condition. They are useful in scenarios where the program waits for an external event or depends on a break condition defined within the loop body.
+
+Motoko uses the `loop` keyword to define an infinite loop. To exit such a loop, you must use a `break` statement with a label, indicating exactly which loop to terminate.
+
+```motoko no-repl
+func unconditionalLoop() : Nat {
+  var count : Nat = 0;
+
+  label countLoop loop {
+    count += 1;
+    if (count == 5) {
+      break countLoop;
+    };
+  };
+  return count;
+}
+```
+
+## `for` loop
+
+A `for` loop in Motoko is used to iterate over the elements of a collection using the following structure:
+
+```motoko no-repl
+for (pattern in iterator) {
+  // Code to run for each item
+}
+```
+
+`for` loops must follow these parameters:
+
+- The iterator must provide a `next()` function that returns either a value (`?T`) or `null` when no items remain.
+
+- The pattern is a variable that matches the type of each item returned by the iterator.
+
+The `for` loop's iterator is evaluated once at the start. Each time through the loop, `next()` is called:
+
+- If it returns a value, it is assigned to pattern, and the loop body runs.
+
+- If it returns `null`, the loop stops.
+
+If evaluating the iterator causes a trap (error), the loop stops immediately.
+
+```motoko no-repl
+let numbers = [1, 2, 3, 4, 5];
+
+for (num in numbers.vals()) {
+  Debug.print(debug_show(num));
+};
+```
```

Comment:
````suggestion

```motoko no-repl
let numbers = [1, 2, 3, 4, 5];

for (num in numbers.vals()) {
  Debug.print(debug_show(num));
};
```

The pattern can also match on values, for example:

```motoko no-repl
let pairs = [(1, 2), (3, 4)];

for ((fst, snd) in pairs.vals()) {
  Debug.print(debug_show(fst + snd));
};
```

````